src/app/core/services/transaction.service.ts

```
@@ -195,17 +195,22 @@
-          const fileIds = fileObjs.map((fileObj) => fileObj.id);
-          if (fileIds.length > 0) {
-            return this.expensesService
-              .createFromFile(fileIds[0], txn.source)
-              .pipe(map((result) => this.transformExpense(result.data[0]).tx));
-          }
+        const fileIds = fileObjs.map((fileObj) => fileObj.id);
+        if (fileIds.length > 0) {
+          return this.expensesService.createFromFile(fileIds[0], txn.source).pipe(
+            switchMap((result) => {
+              // txn contains only source key when capturing receipt
+              if (txn.hasOwnProperty('source') && Object.keys(txn).length === 1) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)